include/umf/pools/pool_disjoint.h

```diff
@@ -58,6 +58,9 @@ typedef struct umf_disjoint_pool_params_t {
 
     /// Name used in traces
     const char *Name;
+
+    /// Whether the pool is limited to host memory
+    int HostMemory;
```

Comment:
this should be some generics bit flags that we can add to.

Reply:
I've disabled poisoning always in DisjointPool for now and created an issue for discussion on the way the optional memory poisoning should be implemented: #634